src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -266,10 +274,6 @@ def _search_role_assignments(scope, assignee, role, include_inherited, include_g
 def _build_role_scope(resource_group_name, resource_id, subscription_id):
     subscription_scope = '/subscriptions/' + subscription_id
     if resource_id:
-        if not subscription_id.lower() in resource_id.lower():
-            # remove when https://github.com/Azure/azure-cli/issues/807 is fixed
-            raise CLIError("Managing assignments on other subscriptions is not yet supported. "
-                           "Run 'az account set' first")
```

Comment:
could we implement the generic --subscription flag instead of just implementing for role?